### DISCUSSION TITLE
(python2 / python3) Unconventional default installation directory

### DISCUSSION THREAD

**Discussion by Marcono1234** [chocolatey-community/chocolatey-packages#1819] [Q&A] [answered]
Currently the default installation directory of python2 ([installer script](https://github.com/chocolatey-community/chocolatey-packages/blob/cbd3e03ae75ffa73171a618a43a9bc84da405092/automatic/python2/tools/chocolateyInstall.ps1#L4)) and python3 ([installer script](https://github.com/chocolatey-community/chocolatey-packages/blob/cbd3e03ae75ffa73171a618a43a9bc84da405092/automatic/python3/tools/chocolateyInstall.ps1#L8)) is unconventional. They are installed directly under the system drive, e.g. as `C:\Python310`.

I would expect it to be installed by default under the conventional `C:\Program Files` (or possibly `C:\Program Files (x86)`), or in the default directory of Chocolately [described in the documentation](https://docs.chocolatey.org/en-us/faqs#where-does-chocolatey-install-by-default).

Or is there are reason why users specifically want Python installed under the system drive (see comments on #106)?

Related: #106, #303

**Comment by AdmiringWorm** [comment, marked as answer]
The location used is due to historical reasons, no other reason.

There was a time that the installer installed to those directories by default, and to keep the package backwards compatible with that and because that was the convention in the beginning it was decided to continue using these directories instead of relying on the default location the installer now uses.

This is also why the package includes the package parameter `/InstallDir:` that can be used to specify the location if you want to use program files instead (it can not be used on the `python` package though, and only on the `python3` and `python2` packages).

**Reply by Marcono1234** [reply]
Thank you for the answer. So there is no plan to change this default installation directory, to remain backwards compatible, correct?
